test/qml/tst_editorwidgets.qml

```diff
@@ -2,45 +2,356 @@ import QtQuick 2.3
+   * setting its value to "six", verifying that it contains "six", resetting its config to disable multiline and html,
+   * setting its value to "seven", and comparing its children's texts with "six" and "seven".
+   */
+  function test_01_textEdit() {
+    const textReadonlyValue = textEdit.children[0]
+    const textField = textEdit.children[1]
+    const textArea = textEdit.children[2]
+
+    compare(textReadonlyValue.text, "") // NOTE: If the config is undefined, the label will be an empty string.
+    compare(textField.text, "DEFAULT_VALUE")
+    compare(textArea.text, "DEFAULT_VALUE")
+
+    textEdit.config = {
+      "IsMultiline": true,
+      "UseHtml": true
+    }
+    textEdit.value = "SECOND_VALUE"
+
+    verify(textReadonlyValue.text.search("SECOND_VALUE") !== -1)
+    compare(textField.text, "SECOND_VALUE")
+    verify(textArea.text.search("SECOND_VALUE") !== -1)
+
+    textEdit.config = {
+      "IsMultiline": false,
+      "UseHtml": false
+    }
+    textEdit.value = "THIRD_VALUE"
+
+    compare(textReadonlyValue.text, "SECOND_VALUE") // NOTE: If the values in the config are set to `false`, the label text will not change.
+    compare(textField.text, "THIRD_VALUE")
+    compare(textArea.text, "THIRD_VALUE")
+  }
+
+
+  /**
+   * Test case for range widget
+   *
+   * This function tests the range widget's functionality by:
+   *
+   * - Verifying that the initial value of the range is correct
+   * - Setting the config to undefined for all properties and verifying that the default values are used
+   * - Verifying that the widget style changes to "TextField" when no style is specified in the config
+   * - Verifying that the precision, min, max, step, and suffix values are set to default values when not specified in the config
+   *
+   * Then, setting the config with specific values and verifying that they are used:
+   *
+   * - Setting the widget style to "Slider" and verifying that it changes
+   * - Setting precision, min, max, step, and suffix values and verifying that they are used
+   *
+   * Finally, setting the value of the range and verifying that it is displayed correctly in both text and slider formats.
+   */
+  function test_01_range() {
+    const sliderRow = range.children[0]
+    const textField = sliderRow.children[0]
+    const valueLabel = range.children[1].children[0]
+    const slider = range.children[1].children[1]
+
+    compare(textField.text, range.default_value + "")
+
+    range.config = {
+      "Style": undefined,
+      "Precision": undefined,
+      "Min": undefined,
+      "Max": undefined,
+      "Step": undefined,
+      "Suffix": undefined
+    }
+    range.value = 3
+    compare(range.widgetStyle, "TextField")
+    compare(range.precision, 1)
+    compare(range.min, -Infinity)
+    compare(range.max, Infinity)
+    compare(range.step, 1)
+    compare(range.suffix, "")
+
+    // Row
+    // compare(sliderRow.visible, true) // ERROR ? should work but not working!
```

Comment:
Can we remove for now until we can either fix the logic or fix what's wrong? :)

Reply:
lets keep it to not forget, and comeback later ?